oracle.go

```diff
@@ -280,6 +280,72 @@ func (o *Oracle) RotateRootCredentials(ctx context.Context, statements []string)
 	return o.RawConfig, nil
 }
 
+func (o *Oracle) SetCredentials(ctx context.Context, statements dbplugin.Statements, staticUser dbplugin.StaticUserConfig) (username, password string, err error) {
+	rotateStatements := statements.Rotation
+	if len(rotateStatements) == 0 {
+		rotateStatements = []string{`ALTER USER "{{username}}" IDENTIFIED BY "{{password}}"`}
+	}
+
+	username = staticUser.Username
+	password = staticUser.Password
+	if username == "" || password == "" {
+		return "", "", errors.New("must provide both username and password")
+	}
+
+	variables := map[string]string{
+		"username": strings.ToUpper(username),
+		"password": password,
+	}
+
+	queries := splitQueries(rotateStatements)
+	if len(queries) == 0 { // Extra check to protect against future changes
+		return "", "", errors.New("no rotation queries found")
+	}
+
+	// Lock the SQL connection
+	o.Lock()
+	defer o.Unlock()
```

Comment:
Question here. I see that this lock already existed. I was wondering if you happen to know why we're locking. Is it because the client isn't thread-safe? Or because the transactions aren't idempotent? Or something else? 

I'm asking because I'd noticed that, on a different client I was looking at (the new Mongo one), it actually did appear thread-safe and handled a connection pool internally.

Reply:
I'm not 100% sure as I made this match what postgres & mysql were already doing. It looks like the Mongo one is also doing it at the beginning of `CreateUser`, `SetCredentials`, & `RevokeUser`. I think it's there to prevent multiple commands from hitting the database in question at the same time.

I'm not entirely sure why that's useful since the database should be able to handle that and in the case of the SQL plugins they're being done in transactions. Perhaps it's to prevent a race between Vault and the database since both need to store the password and stay in sync?

Reply:
Yes, I agree with your thoughts, I'm not sure either. Well, it's certainly harmless to lock, though it could slow things down. I don't mind starting out this way to be safe, and if anyone requests speeding it up we can poke at it more deeply. Thank you!

Reply:
Agreed. If this proves to be a bottleneck we can do some profiling of it.